packages/block-editor/src/store/hooks.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Retrieves the selected block clientId
+ *
+ * @return {string} Block clientId.
+ */
+export function useSelectedBlockClientId() {
+	const clientId = useSelect( ( select ) => {
+		const { getSelectedBlockClientId } = select( 'core/block-editor' );
+		return getSelectedBlockClientId();
+	}, [] );
+
+	return clientId;
+}
+
+/**
+ * Retrieves setAttributes callback for the selected block.
+ *
+ * @return {Function} setAttributes callback.
+ */
+export function useSetAttributes() {
+	const clientId = useSelectedBlockClientId();
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+
+	const setAttributes = useCallback(
+		( newAttributes ) => {
+			updateBlockAttributes( clientId, newAttributes );
+		},
+		[ clientId ]
+	);
+
+	return setAttributes;
+}
+
+/**
+ * Retrieves the selected block attributes
+ *
+ * @return {Object} Block attributes.
+ */
+export function useBlockAttributes() {
+	const clientId = useSelectedBlockClientId();
```

Comment:
While this pattern using hooks inside other hooks is nice and makes the code very readable, I think it has some performance cost. For example, on each useBlockAttributes call we are going to subscribe to the store in useBlockAttributes and in useSelectedBlockClientId while if we did not use useSelectedBlockClientId in useBlockAttributes and just called the selector getSelectedBlockClientId in useBlockAttributes we would have only one subscription.
Is this something we should worry about or the advantages of having code more readable worth the addition store subscriptions? cc: @aduth, @youknowriad